Assets/_Main/Scripts/UI/PlayerHealthUIManager.cs

```diff
@@ -0,0 +1,146 @@
+using System.Collections.Generic;
+using UnityEngine;
+using UnityEngine.UI;
+using System;
+
+/// <summary>
+/// --Health UI Manager--<para></para>
+/// 
+/// This script controls the UI for the health of the player
+/// Also has the upper limit for the number of hearts
+/// Its subcribed to the player events and updates the UI accordingly
+/// </summary>
+/// <remarks> Execution order should be after PlayerHealth. </remarks>
+/// 
+public class PlayerHealthUIManager : MonoBehaviour
+{
+    #region FIELDS
+
+    [Header("Player Data")]
+    [Tooltip("Singleton PlayerHealth script")]
+    [SerializeField] private PlayerHealth _playerHealth;
+
+    [Header("Hearts")]
+    [Tooltip("Gameobject prefab with the image of the full heart")]
+    [SerializeField] private GameObject _heart;
+    [SerializeField] private Image _heartImage;
+    [Tooltip("Gameobject prefab with the image of the hollow heart")]
+    [SerializeField] private GameObject _hollowHeart;
+    [SerializeField] private Image _hollowHeartImage;
+
+    private struct HeartUI // "Ivan" Como se nombran los structs
+    {
+        public GameObject _heartGO;
+        public Image _heartIMG;
+        public Animator _heartAnim;
+    }
+
+    private class HeartDictionary : Dictionary<int, HeartUI> // "Ivan" Como se nombran los diccionaries
+    {
+        private HeartUI newHeart;
+        internal void Add(int key, GameObject heartGO)
+        {
+            newHeart._heartGO = heartGO;
+            newHeart._heartIMG = heartGO.GetComponent<Image>();
+            newHeart._heartAnim = heartGO.GetComponent<Animator>();
+            this.Add(key, newHeart);
+        }
+    }
+
+    HeartDictionary _heartDictionary=new HeartDictionary();
+
+    #endregion
+
+    #region METHODS
+
+    /*
+    private void UpdateHeartUI()
+    {
+        // Podr�a revisar vida m�xima, vida actual y poner los corazones
+    }*/
+
+    void TakeDamage() // "Ivan" En lugar de hacer todo esto es mejor un update heartsUI que revise y los haga todos???
+    {
+        _heartDictionary[_playerHealth.health]._heartAnim.enabled = false;
+        _heartDictionary[_playerHealth.health]._heartIMG.sprite = _hollowHeartImage.sprite;
+    }
+
+    void HealHearts()
+    {
+        int HeartToChange = _playerHealth.health - 1;
+        _heartDictionary[HeartToChange]._heartAnim.enabled = true;
+    }
+
+    void AddHearts()
+    {
+        int HeartToChange = _playerHealth.maxHealth - 1; // "Ivan" Usar constante del player?? crear nueva o hacerla variable para no usar -1 ni --
```

Comment:
Local variables should be camel case.

Reply:
What I would do is to make functions that accept any number, and if they are only going one by one is fine, but it will already be built for more damage.

Reply:
That means you need to add a parameter for all the methods like this one.